prow/plugins/cherrypickunapproved/cherrypick-unapproved.go

```diff
@@ -0,0 +1,151 @@
+		Config: map[string]string{
+			"": fmt.Sprintf(
+				"The cherry-pick-unapproved plugin treats PRs against branch names satisfying the regular expression `%s` as cherry-pick PRs and adds the following comment:\n%s",
+				config.CherryPickUnapproved.BranchRegexp,
+				config.CherryPickUnapproved.Comment,
+			),
+		},
+	}
+	return pluginHelp, nil
+}
+
+type githubClient interface {
+	BotName() (string, error)
+	CreateComment(owner, repo string, number int, comment string) error
+	AddLabel(owner, repo string, number int, label string) error
+	RemoveLabel(owner, repo string, number int, label string) error
+	GetIssueLabels(org, repo string, number int) ([]github.Label, error)
+}
+
+type commentPruner interface {
+	PruneComments(shouldPrune func(github.IssueComment) bool)
+}
+
+func handlePullRequest(pc plugins.PluginClient, pr github.PullRequestEvent) error {
+	return handlePR(pc.GitHubClient, pc.Logger, &pr, pc.CommentPruner, pc.PluginConfig.CherryPickUnapproved.BranchRe, pc.PluginConfig.CherryPickUnapproved.Comment)
+}
+
+func handlePR(gc githubClient, log *logrus.Entry, pr *github.PullRequestEvent, cp commentPruner, branchRe *regexp.Regexp, comment string) error {
+	// Only consider the events that indicate opening of the PR and
+	// when the cpApproved and cpUnapproved labels are added or removed
+	cpLabelUpdated := (pr.Action == github.PullRequestActionLabeled || pr.Action == github.PullRequestActionUnlabeled) &&
+		(pr.Label.Name == cpApprovedLabel || pr.Label.Name == cpUnapprovedLabel)
+	if pr.Action != github.PullRequestActionOpened && pr.Action != github.PullRequestActionReopened && !cpLabelUpdated {
+		return nil
+	}
+
+	var (
+		org    = pr.Repo.Owner.Login
+		repo   = pr.Repo.Name
+		branch = pr.PullRequest.Base.Ref
+		body   = fmt.Sprintf(comment, cpApprovedLabel, cpUnapprovedLabel, branch)
+	)
+
+	// if the branch doesn't match against the branch names allowed for cherry-picks,
+	// don't do anything
+	if !branchRe.MatchString(branch) {
+		return nil
+	}
+
+	labels, err := gc.GetIssueLabels(org, repo, pr.Number)
+	if err != nil {
+		log.WithError(err).Errorf("failed to list labels")
+		return err
+	}
+	hasCherryPickApprovedLabel := github.HasLabel(cpApprovedLabel, labels)
+	hasCherryPickUnapprovedLabel := github.HasLabel(cpUnapprovedLabel, labels)
+
+	// if it has the approved label,
+	// remove the unapproved label (if it exists) and
+	// remove any comments left by this plugin
+	if hasCherryPickApprovedLabel {
+		if hasCherryPickUnapprovedLabel {
+			if err := gc.RemoveLabel(org, repo, pr.Number, cpUnapprovedLabel); err != nil {
+				log.WithError(err).Errorf("Github failed to remove the following label: %s", cpUnapprovedLabel)
+			}
+		}
+
+		botName, err := gc.BotName()
+		if err != nil {
+			return fmt.Errorf("error getting bot name: %v", err)
+		}
+		cp.PruneComments(shouldPrune(log, botName, body))
```

Comment:
The BotName check is unnecessary, `PruneComments` only considers bot comments already.